### PR TITLE
Knative traffic splitting

### LINKED ISSUE #7788: Create custom properties for Knative generation
**Description**

Currently, if you want to customize the output generated by Dekorate for Knative, you only have access to the same properties as in the Kubernetes (https://github.com/quarkusio/quarkus/blob/master/docs/src/main/asciidoc/kubernetes.adoc#knative) but with knative prefix.

The thing is that minimal customizations specific to knative service should be provided, the list is in https://knative.dev/docs/serving/spec/knative-api-specification-1.0/#service I know that not all of them should be implemented, but at least some of most used would be interesting to provide it.

**Implementation ideas**

Implement custom Knative serving attributes to be able to customize the output.

### PR DESCRIPTION
**Pull request by iocanel**
Fixes: #7788

The pull request introduces options to set traffic splitting configuration.

### CODE REVIEW

**Comment by iocanel** [comment]
cc @lordofthejars

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/RevisionTemplateConfig.java [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+import java.util.Map;
+import java.util.Optional;
+
+import io.dekorate.kubernetes.annotation.ImagePullPolicy;
+import io.dekorate.kubernetes.annotation.ServiceType;
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+
+@ConfigGroup
+public class RevisionTemplateConfig {
```
Comment (geoand) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/RevisionTemplateConfig.java:13]: Where is this class being used? I can't seem to find it anywhere else

Comment (iocanel) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/RevisionTemplateConfig.java:13]: Nowhere, I will remove it!

**Review by geoand — APPROVED** [review]

**Comment by iocanel** [comment]
@geoand: ci passed, shall we merge?

**Comment by geoand** [comment]
Yup, feel free

**Pull request merged**